app/src/main/java/framgia/com/ichat/screen/signup/SignUpActivity.java

```diff
@@ -0,0 +1,83 @@
+package framgia.com.ichat.screen.signup;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseActivity;
+
+public class SignUpActivity extends BaseActivity implements SignUpContract.View, View.OnClickListener {
+    private EditText mEditName;
+    private EditText mEditEmail;
+    private EditText mEditPassword;
+    private FirebaseAuth mAuth;
+    private SignUpContract.Presenter mSingUpPresenter;
```

Comment:
Nếu chỉ có 1 presenter trong 1 view ntn e có thể viết ngắn gọn
```suggestion
    private SignUpContract.Presenter mPresenter;
```